diff --git a/node_modules/mobx-react/dist/mobxreact.esm.js b/node_modules/mobx-react/dist/mobxreact.esm.js
index f3ad013..9450838 100644
--- a/node_modules/mobx-react/dist/mobxreact.esm.js
+++ b/node_modules/mobx-react/dist/mobxreact.esm.js
@@ -1,7 +1,7 @@
 import { Reaction, _allowStateChanges, _allowStateReadsStart, _allowStateReadsEnd, $mobx, createAtom, untracked, isObservableMap, isObservableObject, isObservableArray, observable } from 'mobx';
 import React__default, { PureComponent, Component, forwardRef, memo, createElement } from 'react';
 import { isUsingStaticRendering, Observer, observer as observer$1 } from 'mobx-react-lite';
-export { Observer, isObserverBatched, isUsingStaticRendering, observerBatching, observerBatchingOptOut, useAsObservableSource, useLocalStore, useObserver, useStaticRendering } from 'mobx-react-lite';
+export { Observer, isObserverBatched, isUsingStaticRendering, observerBatching, useAsObservableSource, useLocalStore, useObserver, useStaticRendering } from 'mobx-react-lite';
 
 var symbolId = 0;
 
