--- a/IGC/common/SIPKernels/sip.py
+++ b/IGC/common/SIPKernels/sip.py
@@ -79,8 +79,8 @@
         if os.path.exists(sipBin):
             sipBinArray = open(sipBin,"rb").read()
             sipHeaderFile = open(sipHeader,"w+")
-            print("Found "+sipBin)
-            print("Converting "+str(sipBin)+" to "+str(sipHeader))
+            print("Found "+str(sipBin))
+            print("Converting "+str(sipBin)+" to "+str(sipHeader))
             sipHeaderFile.write("namespace SIP\n")
             sipHeaderFile.write("{\n")
             sipHeaderFile.write("    const unsigned char "+sipCharArrayName+"[] =\n")
@@ -96,7 +96,7 @@
             print("Conversion  done.. \n")
             sipHeaderFile.close()
         else:
-            print("Error : Unable to open "+sipInput)
+            print("Error : Unable to open "+str(sipInput))
     else:
         print("Usage sip.py sip.bin sip.h sipStruct")
     return
