--- a/Eigen/src/Core/Matrix.h
+++ b/Eigen/src/Core/Matrix.h
@@ -444,6 +444,7 @@
 EIGEN_MAKE_FIXED_TYPEDEFS(Type, TypeSuffix, 3) \
 EIGEN_MAKE_FIXED_TYPEDEFS(Type, TypeSuffix, 4)

+EIGEN_MAKE_TYPEDEFS_ALL_SIZES(unsigned int,         u)
 EIGEN_MAKE_TYPEDEFS_ALL_SIZES(int,                  i)
 EIGEN_MAKE_TYPEDEFS_ALL_SIZES(float,                f)
 EIGEN_MAKE_TYPEDEFS_ALL_SIZES(double,               d)
